website/static/js/registrationUtils.js

```diff
@@ -419,24 +713,82 @@ var RegistrationEditor = function(urls, editorId) {
     });
     self.currentPage.subscribe(function(page) {
         // lazily apply subscriptions to question values
-        $.each(page.questions(), function(_, question) {
+        $.each(page.questions, function(_, question) {
             question.value.subscribe(function() {
                 self.dirtyCount(self.dirtyCount() + 1);
             });
         });
+        page.viewComments();
     });
 
-    self.canRegister = ko.computed(function() {
-        var draft = self.draft();
-        return draft && draft.isApproved;
+    self.validationErrors = ko.computed(function() {
+        if (self.onLastPage()) {
+            var errors = [];
+            var questions = self.flatQuestions() || [];
+            if (questions.length && questions.filter(function(question) {
+                return question.required && !question.isComplete();
+            }).length) {
+                return 'Some required questions are unanswered.';
+            }
+            else {
+                return '';
+            }
+        }
+        return '';
     });
 
-    self.iterObject = $osf.iterObject;
+    self.canSubmit = ko.computed(function() {
+        var canSubmit = true;
+        var questions = self.flatQuestions();
+        for (var i = 0; i < questions.length; i++) {
+            var question = questions[i];
+            canSubmit = !question.required || (question.required && question.isComplete());
+            if (!canSubmit) {
+                break;
+            }
+        }
+        return canSubmit;
+    });
 
+    self.iterObject = $osf.iterObject;
     // TODO: better extensions system?
     self.extensions = {
-        'osf-upload': editorExtensions.Uploader
+        'osf-upload': editorExtensions.Uploader,
+        'osf-author-import': editorExtensions.AuthorImport
     };
+
+    preview = preview || false;
+    if (preview) {
+        ko.bindingHandlers.previewQuestion = {
+            init: function(elem, valueAccessor) {
+                var question = valueAccessor();
+                var $elem = $(elem);
+
+                if (question.type === 'object') {
+                    $elem.append(
+                        $.map(question.properties, function(subQuestion) {
+                            subQuestion = self.context(subQuestion);
+                            var value;
+                            if (self.extensions[subQuestion.type] ) {
+                                value = subQuestion.preview();
+                            } else {
+                                value = subQuestion.value();
+                            }
+                            return $('<p>').append(value);
```

Comment:
This appears to be a case of KO constructing raw HTML.

Am I correct in assuming that this text contains no user user-entered input of any kind, and sources its value entirely from the premade JSON prereg-schema files?

If this field can contain user input, use `$osf.htmlEscape` or similar.